Competition2017/src/competition/subsystems/autonomous/ShootAndDriveToHopperCommandGroup.java

```diff
@@ -0,0 +1,54 @@
+package competition.subsystems.autonomous;
+
+import com.google.inject.Provider;
+
+import competition.subsystems.drive.commands.DriveForDistanceCommand;
+import competition.subsystems.drive.commands.RotateToHeadingCommand;
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
+import edu.wpi.first.wpilibj.command.CommandGroup;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+
+public class ShootAndDriveToHopperCommandGroup extends CommandGroup {
+    private ShootAndDriveAcrossBaseLineCommandGroup shootAndBaseLine;
+    private DriveForDistanceCommand driveToHopper;
+    private RotateToHeadingCommand rotateToHopper;
+
+    private final DoubleProperty redAllianceHeadingToHopper;
+    private final DoubleProperty blueAllianceHeadingToHopper;
+    private final DoubleProperty distanceToHopperFromTurningPoint;
+    
+    private Alliance color;
+
+    public ShootAndDriveToHopperCommandGroup(XPropertyManager propManager,
+            ShootAndDriveAcrossBaseLineCommandGroup shootAndBaseLine,
+            Provider<DriveForDistanceCommand> driveForDistanceProvider,
+            Provider<RotateToHeadingCommand> rotateToHopperProvider) {
+        redAllianceHeadingToHopper = propManager.createPersistentProperty(" red alliance heading to face hopper", 90);
```

Comment:
Remove preceding space and capitalize. Also, is there a reason we want this to be a property? I recall using a constant the last time we needed hopper headings.